### ISSUE TITLE
FID score comparison

### ISSUE THREAD

**Issue by kwotsin** [TAMU-VITA/AutoGAN#3]
Hi, thanks for providing the code and the discovered architecture looks very interesting to experiment and sheds some insights on how generators could be well designed. In particular, I'm very intrigued by how AutoGAN exploits using upsample-convs rather than transposed convs.

To clarify with you, is the model definition (cifar10_a) at [1] the one that reproduces the best AutoGAN results reported?

Furthermore, could I check if your FID score reported uses 50k real and 50k fake images to compute, since the official FID statistics uses 50k examples (all training data) and your config at [2] uses 50k images by default as well? It seems the other works compared uses different number of images to compute (e.g. SNGAN uses 10k real, 5k fake, refer to pg 16, appendix B.1 of paper [3]). As I see the hyperparameters used compared to SNGAN is quite similar, have you tried computing the AutoGAN's FID score using 10k-real, 5k-fake images for a comparison?

For example, I have run the official SNGAN code [4] and tried using 50k real and 50k fake images to compute FID (using official code and stats from TTUR repo) as well, and it seems that the FID score I got is around 15 (compared to 21.7 for 10k-5k FID score) for unconditional SNGAN.

Thanks for the help! :+1: :+1: 

[1] https://github.com/TAMU-VITA/AutoGAN/blob/master/models/autogan_cifar10_a.py
[2] https://github.com/TAMU-VITA/AutoGAN/blob/master/cfg.py#L143
[3] https://arxiv.org/pdf/1802.05957.pdf
[4] https://github.com/pfnet-research/sngan_projection

**Comment by GongXinyuu** [comment]
Hi kwotsin,

Thank you for your interest on our work. I'd like to clarify with you.

1. Yes, cifar10_a is the top-1 model reported in our paper.

2. We follow TTUR's setting to use 50k fake images and 50k real (all training data) to compute the FID score on CIFAR-10 image generation task. I haven't follow sngan's setting to calculate FID score. I think more sample images may provide better moments approximation for the model distribution, leading to a more accurate FID score. TTUR also indicates that to use a minimum sample size of 10k to calculate the FID otherwise the true FID of the generator is underestimated.

**Comment by GongXinyuu** [comment]
Please let me know if you still have other questions.

**Comment by kwotsin** [comment]
Hi Xinyu,

Thanks for the quick response. Indeed the FID score could be better approximated with 50k-50k as suggested in the original TTUR repository.

Just to check, I have looked at the discriminator for `autogan_cifar10_a` and it seems to be the same one used in SNGAN, so it safe to say they are identical (perhaps except for the initialisations)? This will be interesting since the focus is now on the generator when comparing to SNGAN. For the reported FID score, could I ask how many random seed runs was used (and if there is a standard deviation of FID score you could share)? 

Finally, would it be possible to obtain the pre-trained models for the best performing model? I think it is very interesting upsampling-conv works better than transposed convs, and from some past experiments when I use upsampling (versus max unpool) for shortcuts there are some improvements too, so it's nice to see some confirmation from your work. 

Thank you again.

**Comment by GongXinyuu** [comment]
Hi kwotsin,

Yes, the discriminator is borrowed from sngan. I only use one random seed to run for the FID, which is specified by the config file [cfg.py](https://github.com/TAMU-VITA/AutoGAN/blob/7c6ce06372798c5277c1183e2e6b6e80ca9a6fcb/cfg.py#L149).  

The pre-trained models are available at [Google Drive](https://drive.google.com/drive/folders/1IYDNrKY3m97K3bx_uIzOL6vFmCjGNpYZ?usp=sharing). Please follow the instructions in README to run evaluation.
